### PR TITLE
Add new task helm-render-manifests-from-repo

### PR DESCRIPTION
**Pull request by mikecroft**
# Changes

This task is designed to be used as part of an implementation of the GitOps [Rendered Manifests Pattern](https://akuity.io/blog/the-rendered-manifests-pattern/).

It is intended to be unopinionated and as such will simply template a Helm chart and output a file to a workspace volume with an optionally specified name and location.

It largely follows the same structure as the existing `helm-upgrade-from-repo` task for the sake of consistency. The `OWNERS` file is empty at the moment, I wasn't sure who would be the best person yet.

I am also assuming that no corresponding issue is needed, but I'm happy to create one if there needs to be some discussion around the implementation.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations][authoring]
- [ ] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [ ] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [ ] Commit messages follow [commit message best practices][commit]
- [ ] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/helm-render-manifests-from-repo/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+An example `Pipeline` with a `PipelineRun` can be found in the subdirectory `tests`.
+
+### TaskRun
+
+This `TaskRun` runs the task to set up a Helm repository and generate a template. Validation is disabled in this example.
+
+```yaml
+# example rendered manifests from repo
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/helm-render-manifests-from-repo/0.1/README.md:45]: can this be `tekton.dev/v1` ?

File: task/helm-render-manifests-from-repo/0.1/helm-render-manifests-from-repo.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# This Task will do a helm upgrade based on the given helm repo and chart
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: helm-render-manifests-from-repo
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/categories: Deployment
+    tekton.dev/pipelines.minVersion: "0.12.1"
```
Comment (vinamra28) [inline task/helm-render-manifests-from-repo/0.1/helm-render-manifests-from-repo.yaml:10]: `tekton.dev/v1` was introduced in `v0.50.0`, can we fix the version here as well?

File: task/helm-render-manifests-from-repo/0.1/tests/run.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/helm-render-manifests-from-repo/0.1/tests/run.yaml:2]:
```suggestion
apiVersion: tekton.dev/v1
```

File: task/helm-render-manifests-from-repo/OWNERS [hunk 4]
```diff
@@ -0,0 +1,5 @@
+approvers:
+- 
+
+reviewers:
+-
```
Comment (vinamra28) [inline task/helm-render-manifests-from-repo/OWNERS:5]: are we planning to add someone? else can we remove this?

Comment (mikecroft) [reply task/helm-render-manifests-from-repo/OWNERS:5]: Thanks for the review @vinamra28 - is it OK to not have an `OWNERS` file? That sounds like the quickest resolution, but I'm not sure of the process of adding to the catalog. If it's OK to remove, then I'll make that change along with the other suggestions.

Comment (vinamra28) [reply task/helm-render-manifests-from-repo/OWNERS:5]: yeah, OWNERS file is not mandatory, it's safe to remove

**Comment by mikecroft** [comment]
I've incorporated your suggested changes and made the tests pass @vinamra28 - do you also need me to squash the commits, or is it OK as it is?

**Comment by vinamra28** [comment]
yeah @mikecroft, can you please squash the commits. Will merge this probably post that

**Comment by mikecroft** [comment]
Thanks @vinamra28 - I've squashed the commits and the tests still pass 🥳

**Pull request merged**